src/lib/aws/s3_manager.py

```diff
@@ -32,21 +32,23 @@ def __init__(self, s3_client: boto3.client = None):
         """
         self.s3_client = boto3.client("s3") if s3_client is None else s3_client
 
-    def read_file(self, bucket_name: str, file_name: str) -> str:
+    def read_file(self, s3_path: str) -> str:
         """Read a file from the specified S3 bucket.
 
         Args:
-            bucket_name (str): The name of the S3 bucket.
-            file_name (str): The name of the file to read.
+            s3_path (str): Full S3 path (e.g. s3://your_bucket_name/path/to/your/object/file.txt).
 
         Returns:
             str: The content of the file as a string or None if an error occurs.
 
         """
         try:
-            response = self.s3_client.get_object(Bucket=bucket_name, Key=file_name)
+            s3_path_parts = s3_path.replace("s3://", "").split("/", 1)
```

Comment:
can we use urlparse lib?
https://stackoverflow.com/questions/42641315/s3-urls-get-bucket-name-and-path

the lib is available in lambda

Reply:
done